src/Demo.java

```diff
@@ -1,26 +1,48 @@
+import java.util.ArrayList;
+
 public class Demo {
 
     public static void main(String[] args) {
 
         String word1 = "table";
         String word2 = "tab";
         String word3 = "abe";
```

Comment:
A prerequisite for building easily a prefix tree is to keep the input lexicographically sorted. This one is reverted.